internal/backup/volumesnapshot_action.go

```diff
@@ -113,6 +116,26 @@ func (p *VolumeSnapshotBackupItemAction) Execute(item runtime.Unstructured, back
 		}
 		// save newly applied annotations into the backed-up volumesnapshot item
 		util.AddAnnotations(&vs.ObjectMeta, vals)
+
+		if backupOngoing {
+			p.Log.Infof("Patching volumensnapshotcontent %s with velero BackupNameLabel", vsc.Name)
+			// If we created the volumesnapshotcontent object during this ongoing backup, we would have created it with a DeletionPolicy of Retain.
+			// But, we want to retain these volumesnapsshotcontent ONLY for the lifetime of the backup. To that effect, during velero backup
+			// deletion, we will update the DeletionPolicy of the volumesnapshotcontent and then delete the VolumeSnapshot object which will
+			// cascade delete the volumesnapshotcontent and the associated snapshot in the storage provider (handled by the CSI driver and
+			// the CSI common controller).
+			// However, in the event that the Volumesnapshot object is deleted outside of the backup deletion process, it is possible that
+			// the dynamically created volumesnapshotcontent object will be left as an orphaned and non-discoverable resource in the cluster as well
+			// as in the storage provider. To avoid piling up of such orphaned resources, we will want to discover and delete the dynamically created
+			// volumesnapshotcontents. We do that by adding the "velero.io/backup-name" label on the volumesnapshotcontent.
+			// Further, we want to add this label only on volumesnapshotcontents that were created during an ongoing velero backup.
+
+			pb := []byte(fmt.Sprintf(`{"metadata":{"labels":{"%s":"%s"}}}`, velerov1api.BackupNameLabel, backup.Name))
+			_, vscPatchError = snapshotClient.SnapshotV1beta1().VolumeSnapshotContents().Patch(vsc.Name, types.MergePatchType, pb)
+			if vscPatchError != nil {
+				p.Log.Errorf("Failed to patch volumesnapshotcontent %s: %v", vsc.Name, vscPatchError)
```

Comment:
I am wondering if we should log this as a warning, and not return the error, since it's not critical for the purposes of creating/restoring the backup.

If we return this error (below), then the backup of the PVC will be aborted/it will not be captured in the tarball, which probably *isn't* what we want here.

Reply:
I thought returning this error would result in partial failure of the backup. Meaning, the backup was performed, but there were errors.
If returning an error is going to abort the backup, then that is not what we want. Going to change the log message to a warning and not return this error.

Reply:
it doesn't abort the entire backup, but it does skip backing up the "current item" (i.e. the PVC in this case). See https://github.com/vmware-tanzu/velero/blob/master/pkg/backup/item_backupper.go#L213-L224

Reply:
Thanks for pointing that out! I've updated to log a warning and not return the error.